packages/dx-grid-core/src/plugins/table-band-header/helpers.ts

```diff
@@ -16,24 +16,27 @@ export const isBandedOrHeaderRow: IsSpecificRowFn = tableRow => isBandedTableRow
 export const isNoDataColumn = (columnType: symbol) => columnType !== TABLE_DATA_TYPE;
 
 export const getColumnMeta: GetColumnBandMetaFn = (
-  columnName, bands, tableRowLevel,
+  columnName, bands, tableRowLevel, key = '',
   level = 0, title = null, result = null,
 ) => bands.reduce((acc, column) => {
   if (column.columnName === columnName) {
-    return { ...acc, title, level };
+    return { ...acc, title, level, key };
   }
   if (column.children !== undefined) {
+    const bandTitle = level > tableRowLevel ? title : column.title;
+    const bandKey = level > tableRowLevel ? key : `${key}_${bandTitle}`;
```

Comment:
Should we extract `level > tableRowLevel` into one more variable?

Reply:
Yep, good point